.github/workflows/dotnet-format.yml

```diff
@@ -6,15 +6,41 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout repo
-        uses: actions/checkout@v2
+        uses: actions/checkout@v4
 
-      - name: Add dotnet-format problem matcher
-        uses: xt0rted/dotnet-format-problem-matcher@v1
+      - name: Install .NET 8.0.x
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: "8.0.x"
+
+      - name: Restore Tools
+        run: dotnet tool restore
 
-      - name: Restore dotnet tools
-        uses: xt0rted/dotnet-tool-restore@v1
+      - name: Restore Packages
+        run: dotnet restore osu.Game.Rulesets.Karaoke.sln
 
-      - name: Run dotnet format
-        uses: xt0rted/dotnet-format@v1
+      - name: Restore inspectcode cache
+        uses: actions/cache@v4
         with:
-          only-changed-files: true
+          path: ${{ github.workspace }}/inspectcode
+          key: inspectcode-${{ hashFiles('.config/dotnet-tools.json', '.github/workflows/ci.yml', 'osu.sln*', 'osu*.slnf', '.editorconfig', '.globalconfig', 'CodeAnalysis/*', '**/*.csproj', '**/*.props') }}
+
+      - name: Dotnet code style
+        run: dotnet build -c Debug -warnaserror osu.Game.Rulesets.Karaoke.sln -p:EnforceCodeStyleInBuild=true
```

Comment:
not very sure why `dotnet build -c Debug -warnaserror osu.Game.Rulesets.Karaoke.sln -p:EnforceCodeStyleInBuild=true` works OK in my local WSL
it use the same dotnet version(8.0.411)
maybe can remove or comment out the `-p:EnforceCodeStyleInBuild=true` 

e.g.
```
- name: Dotnet code style
        # run: dotnet build -c Debug -warnaserror osu.Game.Rulesets.Karaoke.sln -p:EnforceCodeStyleInBuild=true
        run: dotnet build -c Debug -warnaserror osu.Game.Rulesets.Karaoke.sln
```

or disable this step and write the reason in the ymal.